### ISSUE TITLE
[Feature Request] Proxy Protocol v2 Support

### ISSUE THREAD

**Issue by Showdown76py** [Starystars67/BeamNG-MP#381]
**Is your feature request related to a problem? Please describe.**
This issue is related to a problem: if a server is working is behind a proxy that is using proxy protocol v2, it will not work and the connection to the socket will be refused. Proxy Protocol v2 proxies allows the server to see the IP from the client.

**Describe the solution you'd like**
An option to toggle in the configuration file, that enables proxy protocol v2 support.

**Describe alternatives you've considered**
There is no other solution than this.

**Comment by OfficialLambdax** [comment]
Wrong repo. This repo here is for the game mod of beammp. Here is the server repo https://github.com/BeamMP/BeamMP-Server

Besides that, i dont think anyone will implement the support of your desired protocol. You can do that yourself tho, the code is in the repo above.

Then id like to add a alternative. You could make your own tiny relay that listens to the incoming connections and that supports your protocol and then forward that connection to the game server, once ready.

**Comment by Showdown76py** [comment]
Sorry, I published this issue too quickly and didn't even properly verify if I was on the correct repo.
Proxy Protocol v2 can be common and I know it is used by some anti-ddos services such as TCPShield, HAProxy (and anything that allows reverse proxying)
I do not have any knowledge in C++ and that will be difficult for me to implement that by myself. Also, I did not really understand the alternative that you have suggested. Thanks!
